### PR TITLE
doc: add use-cases section to readme

### PR DESCRIPTION
**Pull request by nfriedly**
And rework alternate rate limiters section

Should help to address some of the confusion evidenced in https://github.com/express-rate-limit/express-rate-limit/issues/366 and other tickets

Needs formatting because I just did the editing in github's ui

### CODE REVIEW

**Review by gamemaker1 — APPROVED** [review]
LGTM!

**Comment by gamemaker1** [comment]
Should we merge this?

File: readme.md [hunk 1]
```diff
@@ -118,6 +118,7 @@ const limiter = rateLimit({
 	max: 100, // Limit each IP to 100 requests per `window` (here, per 15 minutes)
 	standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
 	legacyHeaders: false, // Disable the `X-RateLimit-*` headers
+	store: new ExternalStore(), // Use an external store for more precise rate limiting
```
Comment (nfriedly) [inline readme.md:121]: Hum, I'm a little iffy about this. I feel like `ExternalStore` is going to confuse some folks. Maybe the redis store?

Comment (nfriedly) [reply readme.md:121]: what about 
```suggestion
	// store: ... // Use an external store for more precise rate limiting
```

File: readme.md [hunk 2]
```diff
@@ -151,15 +178,15 @@ To use a custom store:
 ```ts
 import rateLimit, { MemoryStore } from 'express-rate-limit'
 
-const apiLimiter = rateLimit({
+const rateLimiter = rateLimit({
 	windowMs: 15 * 60 * 1000, // 15 minutes
 	max: 100, // Limit each IP to 100 requests per `window` (here, per 15 minutes)
 	standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
 	store: new MemoryStore(),
```
Comment (nfriedly) [inline readme.md:185]: maybe we should use the redis store here

Comment (gamemaker1) [reply readme.md]: Done

**Comment by nfriedly** [comment]
cool, I'm going to merge this and tag it for release

**Pull request merged**